### DISCUSSION TITLE
@urql/preact in Next projects

### DISCUSSION THREAD

**Discussion by mvarendorff** [urql-graphql/urql#1524] [Q&A]
`@urql/preact` doesn't function in next projects (example repository [here](https://github.com/geisterfurz007/urql-preact-issue-next)). Given that Next is based on React and Preact compatibility is only achieved by aliasing the npm package (both in package.json as well as through webpack), the urql package for React functions properly.

I am honestly not sure why `@urql/preact` fails (see an earlier conversation [here](https://github.com/FormidableLabs/urql/issues/644#issuecomment-813610681)) since the preact package is definitely present (possibly some webpack magic in the next-plugin-preact) but this could possibly be added to the `@urql/preact` docs (or not, if too much of an edgecase) but that's why I opened this discussion.

**Comment by Sporesirius** [comment]
@geisterfurz007 were you able to figure out the problem?
I found a issue and discussion, but it had not really helped : https://github.com/FormidableLabs/urql/issues/1215 https://github.com/FormidableLabs/urql/discussions/1069

**Reply by mvarendorff** [reply]
I haven't had the chance to look into this deeper again yet, unfortunately. I haven't checked whether server side queries function properly but the website is building at least with these dependencies (using urql instead of urql/preact):

https://github.com/geisterfurz007/fullstack-typescript-boilerplate/blob/9edbf1407eff6258702dff6a426d4e59ed9a02a1/packages/web/package.json#L30-L40
